adjudicator/state.py

```diff
@@ -71,25 +71,25 @@ def get_encoding(self, card):
         return self.values[element]
 
     def hide_encoding(self, player):
+        """
+         Mask state vector values
+
+         :param player: the player number 0-3 to recieve a tailored masked encoding state vector
+         :return: masked encoding np array
+         """
         player_1 = [1, 11, 21]
         player_2 = [2, 12, 22]
         player_3 = [3, 13, 23]
         player_4 = [4, 14, 24]
         players = [player_1, player_2, player_3, player_4]
-        ret = self.values
-        nrow = 0
-        ncolumn = 0
-        for row in ret:
-
-            for column in row:
-                if column in players[player]:
-                    continue
-                else:
-                    ret[nrow , ncolumn] = 0
-                ncolumn += 1
-            nrow == 1
-
-        return ret
+        ret = []
+        for x in self.values:
+            if x in players[player]:
+                ret.append(x)
+            else:
+                ret.append(0)
+        return np.array(ret)
+
```

Comment:
with numpy you can do logical masks, which will make this a much simpler task

take a look at the numpy.where function. 

since we only care about masking the hidden information we can create create a boolean vector that return true where the the array contains the value for the player number as well as where the value in the array is greater than 10 ie cards that have been played, and false everywhere else

that will create a vector of 52 true false statements. You can then use this vector in the numpy.where function to select between the values vector where true or an equal sized zeros vector where false. 


for example if I'm doing the same process as player 1 on a 6 element vector

values = [1,1,21,13,2,4]
--- np.array([1,1,21,13,2,4])

mask = values[values == 1] + values[values > 10]
--- np.array([True, True, True, True, False, False])

Reply:
Alright I made the changes.
However after going through the documentation, true indicates that a value was masked and false indicates that a value is not masked. 
So in your example:

values = [1,1,21,13,2,4]

boolean_vector = [False, False, False, True, True]

The results ended up being the same, and I think it could go either way. I just had to play around with the condition statements to do that.

Reply:
Cool no worries, 6 one way half a dozen another. 

This is the code I tested. 

```python
def hide_encoding(player):
    cards_in_hand = self.values == player
    played_cards = self.values > 10

    return np.where(cards_in_hand + played_cards, self.values, np.zeros(52, dtype=int))

Reply:
Wow, that's quite a bit more readable. Also skips unnecessary steps. Could we go with this?

Reply:
of course